### PR TITLE
Core breakdown

### PR DESCRIPTION
**Pull request by adrn**
I broke up the `core.py` module into `observer.py` and `target.py` and made a few tweaks to the imports in other modules to clean up a bit.

### CODE REVIEW

**Comment by cdeil** [comment]
:+1: to `observer.py` and `target.py` to organise the code.

I don't have time to review this, I presume @bmorris3 will.
Just one comment: this will conflict with #82, but this one should probably go first?

**Comment by adrn** [comment]
Right, I fear I may have created a bit more work for @bsipocz :(

File: astroplan/observer.py [hunk 1]
```diff
@@ -247,7 +223,7 @@ def __repr__(self):
     @classmethod
     def at_site(cls, site_name, **kwargs):
         """
-        Initialize an `~astroplan.core.Observer` object with a site name.
+        Initialize an `~astroplan.observer.Observer` object with a site name.
 
         Extra keyword arguments are passed to the `~astroplan.core.Observer`
```
Comment (bmorris3) [inline astroplan/observer.py]: `~astroplan.core.Observer` > `~astroplan.observer.Observer`

File: astroplan/observer.py [hunk 2]
```diff
@@ -247,7 +223,7 @@ def __repr__(self):
...
         constructor (see `~astroplan.core.Observer` for available keyword
```
Comment (bmorris3) [inline astroplan/observer.py]: ~astroplan.core.Observer > ~astroplan.observer.Observer

**Comment by bmorris3** [comment]
I'd like to vote for adding a parenthetical "(hard)" prefix, and a :notes: suffix to this PR.

**Comment by bmorris3** [comment]
Looks good, merging.

**Pull request merged**